iree/compiler/Conversion/Passes.cpp

```diff
@@ -17,6 +17,47 @@ namespace {
 void registerConversionPasses() {
   // Generated.
   registerPasses();
+
+  static PassPipelineRegistration<LLVMTransformPassPipelineOptions>
+      linalgLLVMVPipeline(
+          "iree-codegen-linalg-to-llvm-pipeline",
+          "Runs the progressive lowering pipeline from Linalg to LLVM",
+          [](OpPassManager &passManager,
+             const LLVMTransformPassPipelineOptions &options) {
+            buildLLVMTransformPassPipeline(passManager, options);
+          });
+
+  static PassPipelineRegistration<> LinalgNVVMPipeline(
+      "iree-codegen-linalg-to-nvvm-pipeline",
+      "Runs the progressive lowering pipeline from Linalg to NVVM",
+      [](OpPassManager &passManager) {
+        buildLLVMGPUTransformPassPipeline(passManager, false);
+      });
+
+  static PassPipelineRegistration<> LinalgROCDLPipeline(
+      "iree-codegen-linalg-to-rocdl-pipeline",
+      "Runs the progressive lowering pipeline from Linalg to ROCDL",
+      [](OpPassManager &passManager) {
+        buildLLVMGPUTransformPassPipeline(passManager, true);
+      });
+
+  // TODO: Figure out why this triggers a linkage error.
+  // static PassPipelineRegistration<> linalgToSPIRVPipeline(
+  //     "iree-codegen-linalg-to-spirv-pipeline",
+  //     "Runs the progressive lowering pipeline from Linalg to SPIR-V",
+  //     [](OpPassManager &passManager) {
+  //       buildLinalgToSPIRVPassPipeline(passManager,
+  //                             SPIRVCodegenOptions::getFromCLOptions());
+  //     });
+
+  static PassPipelineRegistration<> hloToLinalgSPIRVPipeline(
```

Comment:
Maybe we can drop this one. This is probably left over stuff from the old buffer path. Its no longer supported. This might help with the linking error above.

Reply:
They are both used and I didn't spend enough time to figure out how/why. I think there is some naming and comment drift that would be good for you to clean up. Left a TODO.